Add literal string support to include and exclude filters (#1068)
python-attrs/attrs: docs/examples.md

>>> asdict(
...     User("jane", "s33kred", 42),

> ...     filter=filters.exclude(fields(User).password, "password", int))

Reviewer: This makes no sense, since it's using password twice which is confusing (at least it's confusing me right now).

Maybe add a field?

Reviewer 2: Maybe `email`?

...     z: int

>>> asdict(C("foo", "2", 3),

> ...        filter=filters.include(int, fields(C).x, "x"))

Reviewer: same as above

Author: Removed. One example probably is enough.

Author: Fine. Add it back. More examples are always better?

{'x': 'foo', 'z': 3}
```

:::{note}

> Though using string names directly is convenient, it's not safe in typo issues.

Reviewer: ```suggestion
Though using string names directly is convenient, mistyping attribute names will silently do the wrong thing and neither Python nor your type checker can help you.
```

> Through using string names directly is convenient, it's not safe in typo issues.

Author: Thanks! 😃

> `fields()` will raise an `AttributeError` when the field doesn't exist while literal string names won't.

Reviewer: ```suggestion
{func}`attrs.fields()` will raise an `AttributeError` when the field doesn't exist while literal string names won't.
```

>>> asdict(C("foo", "2", 3),
...        filter=filters.include(int, fields(C).x))
{'x': 'foo', 'z': 3}

>>> asdict(C("foo", "2", 3),

> ...        filter=filters.include(fields(C).x), "z")

Reviewer 2: Is this a `)` to much?

> ...        filter=filters.include(fields(C).x), "z")

Reviewer 2: ```suggestion
...        "z", filter=filters.include(fields(C).x))
```

Reviewer 2: Yes, you're right.
